spec/datadog/appsec/integration/contrib/rails/endpoint_collection_spec.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+require 'datadog/tracing/contrib/support/spec_helper'
+require 'datadog/appsec/spec_helper'
+require 'rack/test'
+
+require 'action_controller/railtie'
+require 'datadog/tracing'
+require 'datadog/appsec'
+
+RSpec.describe 'Rails Endpoint Collection' do
+  include Rack::Test::Methods
+
+  before do
+    app = Class.new(Rails::Application) do
+      config.root = __dir__
+      config.secret_key_base = 'test-secret-key-base'
+      config.action_dispatch.show_exceptions = :rescuable
+      config.hosts.clear
+      config.eager_load = false
+      config.consider_all_requests_local = true
+      config.logger = Rails.logger = Logger.new(StringIO.new)
+
+      config.file_watcher = Class.new(ActiveSupport::FileUpdateChecker) do
+        def initialize(files, dirs = {}, &block)
+          dirs = dirs.delete('') if dirs.include?('')
+
+          super
+        end
+      end
+    end
+
+    stub_const('RailsTest::Application', app)
+
+    Datadog.configure do |c|
+      c.tracing.enabled = true
+      c.appsec.enabled = true
+      c.appsec.instrument :rails
+
+      c.remote.enabled = false
+    end
+
+    app.initialize!
+
+    app.routes.draw do
+      resources :products
+
+      match '/search', to: 'search#index', via: :all
```

Comment:
Does it make sense to add move common route definitions, including

1. get/post/put/patch
2. namespaces (and conditions, to be sure)
3. resources + only (too see that reduction works)